File: resource-registry/src/main/java/nl/haploid/octowight/registry/configuration/MongoConfiguration.java
package nl.haploid.octowight.registry.configuration;

import com.mongodb.MongoClient;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.mongodb.MongoDbFactory;
import org.springframework.data.mongodb.core.MongoTemplate;
import org.springframework.data.mongodb.core.SimpleMongoDbFactory;
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories;

import java.net.UnknownHostException;

@Configuration
@EnableMongoRepositories(basePackages = "nl.haploid.octowight.registry.repository")
public class MongoConfiguration {

	// TODO: drop IT database after tests

	@Value("${octowight.registry.mongo.database}")
	private String mongoDatabase;

	@Value("${octowight.registry.mongo.hostname}")
	private String mongoHostname;

	@Value("${octowight.registry.mongo.port}")
	private Integer mongoPort;

	public String getMongoDatabase() {
		return mongoDatabase;
	}

	@Bean
	public MongoDbFactory mongoDbFactory() throws UnknownHostException {
		final MongoClient mongoClient = new MongoClient(mongoHostname, mongoPort);
		return new SimpleMongoDbFactory(mongoClient, mongoDatabase);
	}

	@Bean
	public MongoTemplate mongoTemplate(final MongoDbFactory mongoDbFactory) {
		return new MongoTemplate(mongoDbFactory);
	}
}

File: resource-registry/src/main/scala/nl/haploid/octowight/registry/configuration/MongoConfiguration.scala
package nl.haploid.octowight.registry.configuration

import com.mongodb.MongoClient
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.{Bean, Configuration}
import org.springframework.data.mongodb.MongoDbFactory
import org.springframework.data.mongodb.core.{MongoTemplate, SimpleMongoDbFactory}
import org.springframework.data.mongodb.repository.config.EnableMongoRepositories

@Configuration
@EnableMongoRepositories(basePackages = Array("nl.haploid.octowight.registry.repository"))
class MongoConfiguration {

  @Value("${octowight.registry.mongo.database}") private[this] val mongoDatabase: String = null
  @Value("${octowight.registry.mongo.hostname}") private[this] val mongoHostname: String = null
  @Value("${octowight.registry.mongo.port}") private[this] val mongoPort: Integer = null

  def getMongoDatabase = mongoDatabase

  @Bean
  def mongoDbFactory: MongoDbFactory = {
    val mongoClient = new MongoClient(mongoHostname, mongoPort)
    new SimpleMongoDbFactory(mongoClient, mongoDatabase)
  }

  @Bean
  def mongoTemplate(mongoDbFactory: MongoDbFactory) = new MongoTemplate(mongoDbFactory)
}
